client/docs/providers.md

```diff
@@ -0,0 +1,248 @@
+
+[//]: # (|                 [Pinecone]&#40;#pinecone&#41;                 |         |   x    |         |    x    |   x    | x                      |)
+
+[//]: # (|                 [Postgres]&#40;#postgres&#41;                 |    x    |        |    x    |    x    |        |                        |)
+
+[//]: # (|                    [Redis]&#40;#redis&#41;                    |         |   x    |         |    x    |   x    |                        |)
+
+[//]: # (|                 [Redshift]&#40;#redshift&#41;                 |    x    |        |    x    |    x    |        |                        |)
+
+[//]: # (|                       [S3]&#40;#s3&#41;                       |    x    |        |         |    x    |        |                        |)
+
+[//]: # (|                [Snowflake]&#40;#snowflake&#41;                |    x    |        |    x    |    x    |        |                        |)
+
+[//]: # (|                    [Spark]&#40;#spark&#41;                    |    x    |        |    x    |         |        |                        |)
+
+[//]: # (|                 [Weaviate]&#40;#weaviate&#41;                 |         |   x    |         |    x    |   x    | x                      |)
+
+
+## Provider Registration
+This page provides reference and examples for how to register the various providers that FeatureForm supports.
+
+### Azure Blob Store
+::: featureform.register.Registrar.register_blob_store
+    handler: python
+    options:
+        show_root_heading: false
+        show_source: false
+        show_root_toc_entry: false
+
+### BigQuery
+::: featureform.register.Registrar.register_bigquery
+    handler: python
+    options:
+        show_root_heading: false
+        show_source: false
+        show_root_toc_entry: false
+
+### Cassandra
+::: featureform.register.Registrar.register_cassandra
+    handler: python
+    options:
+        show_root_heading: false
+        show_source: false
+        show_root_toc_entry: false
+
+### DynamoDB
+::: featureform.register.Registrar.register_dynamodb
+    handler: python
+    options:
+        show_root_heading: false
+        show_source: false
+        show_root_toc_entry: false
+
+### Firestore
+::: featureform.register.Registrar.register_firestore
+    handler: python
+    options:
+        show_root_heading: false
+        show_source: false
+        show_root_toc_entry: false
+
+### Google Cloud Storage
+::: featureform.register.Registrar.register_gcs
+    handler: python
+    options:
+        show_root_heading: false
+        show_source: false
+        show_root_toc_entry: false
+
+### HDFS
+::: featureform.register.Registrar.register_hdfs
+    handler: python
+    options:
+        show_root_heading: false
+        show_source: false
+        show_root_toc_entry: false
+
+### Kubernetes Pandas Runner
+::: featureform.register.Registrar.register_k8s
```

Comment:
We should specify how to leverage the docker image and to leverage our image as a base image.